File: src/main/java/spring/webmvc/infrastructure/util/SecurityContextUtil.java
package spring.webmvc.infrastructure.util;

import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;

public abstract class SecurityContextUtil {

	public static Long getMemberId() {
		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();

		if (authentication == null || !authentication.isAuthenticated()) {
			throw new BadCredentialsException("인증되지 않은 사용자입니다.");
		}

		return Long.valueOf(String.valueOf(authentication.getPrincipal()));
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/util/SecurityContextUtil.kt
package spring.web.kotlin.infrastructure.util

import org.springframework.security.authentication.BadCredentialsException
import org.springframework.security.core.context.SecurityContextHolder

object SecurityContextUtil {
    fun getMemberId(): Long {
        val authentication = SecurityContextHolder.getContext().authentication

        if (authentication == null || !authentication.isAuthenticated) {
            throw BadCredentialsException("인증되지 않은 사용자입니다.")
        }

        return authentication.principal.toString().toLong()
    }
}